Enable gnome installation with /home partition encrypted

b10n1k opened the pull request:
We perform an AutoYast installation creating a separate partition
for home and we encrypt it. In additional we select Workstation Extension
addon and we install gnome-x11 pattern.
As occurs with encrypted boot partition the installation module stops
after the grub and we handle the decryption in the next module.
We verify the scenario at the end reusing the verify_separate_home
and testing the clone function.


- Related ticket: https://progress.opensuse.org/issues/65172
- Needles: 
- Verification run: http://aquarius.suse.cz/tests/2667
- Test plan: https://gitlab.suse.de/qsf-y/qa-sle-functional-y/-/merge_requests/180

jknphy commented on tests/autoyast/installation.pm:
```diff
@@ -195,6 +195,10 @@ sub run {
         }
+        elsif (match_has_tag('import-untrusted-gpg-key')) {
+            handle_untrusted_gpg_key;
+            next;
```
let's `grep` to exclude the needle before `next` to avoid more times matching and lost keys.

b10n1k replied:
Done

jknphy commented:
I'd suggest to validate that home is encrypted as well.

b10n1k commented:
> I'd suggest to validate that home is encrypted as well.

yea i was thinking about it as well (if we will need another check for it). But then i thought that you encounter the decryption in the step after the installation which is clear indication that the disk is encrypted[0]. notice the cr_home. In addition, i could perform a read/write test. Do you think we really need those or maybe any other?

[0] http://aquarius.suse.cz/tests/2627#step/decrypt_partition/1

jknphy commented:
> > I'd suggest to validate that home is encrypted as well.
> 
> yea i was thinking about it as well (if we will need another check for it). But then i thought that you encounter the decryption in the step after the installation which is clear indication that the disk is encrypted[0]. notice the cr_home. In addition, i could perform a read/write test. Do you think we really need those or maybe any other?
> 
> [0] http://aquarius.suse.cz/tests/2627#step/decrypt_partition/1

yes, we probably implicitly tested, but if other partition is encrypted instead we will miss it, there is some existing module to validate encryption, maybe we could reuse there. Could you check?

Another point, it would be useful here to validate the cloned profile.
A minor thing: `verify_separate` and any other validations by conventions we schedule them the last, if you don't mind to adapt the order.

b10n1k commented:
> yes, we probably implicitly tested, but if other partition is encrypted instead we will miss it, there is some existing module to validate encryption, maybe we could reuse there. Could you check?
> 
I added and i modify the check accordantly.

> Another point, it would be useful here to validate the cloned profile.

Added

> A minor thing: `verify_separate` and any other validations by conventions we schedule them the last, if you don't mind to adapt the order.

bear with me. Every module is a validation. Every validation should run after the SUT is full ready(installed, configured, etc). The modules that you suggest that they are not in order is not part of the installation IMO and do not affect the validation. in this particular one i try to keep the important validation as close as the installation. For me the rest (in this particular scheduling again) are extra checks for additional coverage.

jknphy commented on schedule/yast/autoyast_home_encrypted.yaml:
```diff
@@ -0,0 +1,75 @@
+  - installation/isosize
+  - installation/bootloader_start
+  - autoyast/installation
+  - installation/decrypt_partition
```
can we use here existing module to decrypt? https://openqa.suse.de/tests/4186828#step/boot_encrypt/1

b10n1k replied:
i can and make sense. What i am waiting for is an answer from @OleksandrOrlov because the name of the existing one is boot_encrypt and i wonder if i should rename it to reflect the generalization of the purpose.

jknphy commented:
btw your review appears as 'Draft', how did it get to be like that? do I need to click in 'Ready for review' button for you or you have privileged to do it? I'm not familiar with that feature...

b10n1k commented:
i use 

> btw your review appears as 'Draft', how did it get to be like that? do I need to click in 'Ready for review' button for you or you have privileged to do it? I'm not familiar with that GitHub feature...

'Draft' indicates that it can not be merged if it is not reviewed. Yes i can switch to 'Ready for review' but lets wait to make clear the naming of the `decrypt_partition`.

b10n1k commented:
@OleksandrOrlov Could you review whenever you have the time

jknphy commented on lib/utils.pm:
```diff
@@ -720,7 +720,7 @@ that the boot partition is encrypted.
 =cut
 sub is_boot_encrypted {
-    return 0 if get_var('UNENCRYPTED_BOOT');
+    return 0 if get_var('UNENCRYPTED_BOOT') || get_var('HOME_ENCRYPTED');
```
why we need this change? I'm not sure ...because we are fixing the modules schedules, also is not documented in variables.md either added to schedule in `vars:` section.

b10n1k replied:
FYI i have remove it and use the `UNENCRYPTED_BOOT`

OleksandrOrlov commented on tests/autoyast/installation.pm:
```diff
@@ -345,6 +351,7 @@ sub run {
     push(@needles, 'opensuse-welcome') if opensuse_welcome_applicable;
     # There will be another reboot for IPMI backend
     push @needles, qw(prague-pxe-menu qa-net-selection) if check_var('BACKEND', 'ipmi');
+
```
Could you please remove this empty line from the commit.

b10n1k replied:
Done

OleksandrOrlov commented on lib/utils.pm:
```diff
@@ -720,7 +720,7 @@ that the boot partition is encrypted.
 =cut
 sub is_boot_encrypted {
-    return 0 if get_var('UNENCRYPTED_BOOT');
+    return 0 if get_var('UNENCRYPTED_BOOT') || get_var('HOME_ENCRYPTED');
```
`HOME_ENCRYPTED` var is not used in your yaml scheduling file.

I guess, you can remove it, just set the existing `UNENCRIPTED_BOOT` in scheduling file.

b10n1k replied:
yes  can replace the Variable. but i have set it up in the TestSuite. Should it be in yaml as well?

OleksandrOrlov replied:
You have both `ENCRYPTED` and `HOME_ENCRYPTED` variables in the test suite. 

It seems like `HOME_ENCRYPTED` only needed to return `0` in `boot_encrypted` function. Why not just set already existing variable `UNENCRYPTED_BOOT=1` instead of introducing new one for the same reason?

If you want to keep `HOME_ENCRYPTED`, then the logic here looks questionable for me. How "home" encryption related to the "boot" encryption? May it be the case, when both "boot" and "home" are encrypted?

b10n1k replied:
i have remove it and use the `UNENCRYPTED_BOOT`. it is set in TestSuite

OleksandrOrlov commented on tests/autoyast/installation.pm:
```diff
@@ -196,6 +197,11 @@ sub run {
             send_key 'ret';    # press enter if grub timeout is disabled, like we have in reinstall scenarios
             last;              # if see grub, we get to the second stage, as it appears after bios-boot which we may miss
         }
+        elsif (match_has_tag('import-untrusted-gpg-key')) {
```
I tried to find any other test suite where we have this popup to be appeared, but did not find.

Do you know why this 'untrusted gpg key' popup is appeared at all in your test?

b10n1k replied:
the needle is already in the repo. i cant find any specific match atm but there are some tests that use it

OleksandrOrlov commented on tests/autoyast/installation.pm:
```diff
@@ -131,7 +131,8 @@ sub run {
     test_ayp_url;
     my $test_data = get_test_suite_data();
-    my @needles = qw(bios-boot nonexisting-package reboot-after-installation linuxrc-install-fail scc-invalid-url warning-pop-up autoyast-boot package-notification nvidia-validation-failed);
+    my @needles = qw(bios-boot nonexisting-package reboot-after-installation linuxrc-install-fail scc-invalid-url warning-pop-up autoyast-boot package-notification nvidia-validation-failed import-untrusted-gpg-key);
```
Did you add new 'import-untrusted-gpg-key' needle?

If you've added it just locally, could you please also make a PR to needles repo and paste the link to it.

b10n1k replied:
no this exists already

jknphy commented on lib/utils.pm:
```diff
@@ -579,24 +579,21 @@ sub zypper_ar {
-    my $cmd_mr  = "mr -p $priority $url";
+    $priority     = defined($priority) ? "-p $priority"  : "";
+    $no_gpg_check = $no_gpg_check      ? "--no-gpgcheck" : "";
+    my $cmd_ar  = "--gpg-auto-import-keys ar -f $priority $no_gpg_check $params $url";
```
could you kindly explain why is needed this change in this file for this scenario, I cannot find the reason to touch this library and do it without adding and additional commit for it?

b10n1k replied:
this is not me. i remember doing a rebase after removing my change there. let me check

b10n1k replied:
i reverted it. Sorry about that
